fetch-in-react/my-app/src/UserDetails.tsx

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserCard } from './UserCard';
+import type { User } from './Users';
+
+type Props = {
+  userId: number;
+  onCancel: () => void;
+};
+export function UserDetails({ userId, onCancel }: Props) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [user, setUser] = useState<User>();
+
+  /* your code here (hint: useEffect) */
+
+  useEffect(() => {
+    async function getUserData() {
+      try {
+        const response = await fetch(
+          `https://jsonplaceholder.typicode.com/users/${userId}`
+        );
+        const data = await response.json();
+        if (response.ok) {
+          setUser(data);
+        }
+      } catch (error) {
+        setError(error);
+      }
+    }
+    setIsLoading(false);
```

Comment:
You shouldn't set isLoading to false until after the fetch request completes. This code will set it false as soon as the effect runs. So the Loading... display will disappear almost immediately. You want it to stay up as long as the http request is in process.